sktime/forecasting/mapa.py

```diff
@@ -0,0 +1,770 @@
+from sktime.forecasting.exp_smoothing import ExponentialSmoothing
+from sktime.utils.datetime import _get_freq
+from sktime.utils.validation.series import check_series
+
+
+class MAPAForecaster(BaseForecaster):
+    """MAPA (Multiple Aggregation Prediction Algorithm) Forecaster."""
+
+    def __init__(
+        self,
+        comb: str = "w.mean",
+        hybrid: bool = True,
+        conf_lvl: Optional[float] = None,
+        sp: Optional[int] = None,
+        etstype: str = "ets",  # Add this line
+    ):
+        super().__init__()
+        self.comb = comb
+        self.hybrid = hybrid
+        self.conf_lvl = conf_lvl
+        self.sp = sp
+        self.etstype = etstype  # Add this line
+        self._y = None
+        self._mapafit = None
+        self._is_fitted = False
+
+    @property
+    def y_(self):
+        """Returns _y."""
+        return self._y
+
+    @property
+    def mapafit_(self):
+        """Returns _mapafit."""
+        return self._mapafit
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit forecaster to training data."""
+        self._y = check_series(y).copy()
+        self._mapafit = self._prepare_mapafit(self._y)
+        if X is not None:
+            self._validate_xreg(X, len(fh) if fh is not None else 0, len(self._y))
+        self._is_fitted = True
+        return self
+
+    def _predict(self, fh, X=None):
+        """Make forecasts for the given forecast horizon."""
+        self._check_is_fitted()
+
+        # Create deep copies of internal state to prevent modifications
+        y_copy = self._y.copy()
+        mapafit_copy = deepcopy(self._mapafit)
+
+        result = self.mapafor(
+            y_copy,
+            mapafit_copy,
+            fh=len(fh),
+            comb=self.comb,
+            hybrid=self.hybrid,
+            conf_lvl=self.conf_lvl,
+            xreg=X,
+        )
+
+        if len(result["outfor"]) == 0:
+            return pd.Series(index=fh)  # Return an empty Series with the correct index
+        else:
+            return pd.Series(result["outfor"], index=fh)
+
+    def _update(self, y, X=None, update_params=True):
+        """Update the forecaster with new data."""
+        if not self._is_fitted:
+            return self._fit(y, X)
+
+        self._y = pd.concat([self._y, check_series(y)])
+
+        if update_params:
+            self._mapafit = self._prepare_mapafit(self._y)
+
+        return self
+
+    def _prepare_mapafit(self, y):
+        """Prepare MAPA fit data using AutoETS."""
+        sp = self.sp if self.sp is not None else self._infer_seasonal_period(y)
+
+        return pd.DataFrame(
+            {
+                "use": [True, True],
+                "AL": [1, sp],
+                "original.ppy": [sp, sp],
+                "etstype": [self.etstype, self.etstype],
+            }
+        )
+
+    def _fit_auto_ets(self, yA: pd.Series, ppyA: int) -> AutoETS:
+        """Fit AutoETS model."""
+        auto_ets = AutoETS(sp=ppyA, auto=True)
```

Comment:
I suggest making the forecaster a hyper-parameter, with `AutoETS` as a default (i.e., if the param is `None`)